src/rgw/rgw_data_sync.cc

```diff
@@ -3389,7 +3390,7 @@ RGWCoroutine *RGWRemoteBucketManager::read_sync_status_cr(int num, rgw_bucket_sh
     return nullptr;
   }
 
-  return new RGWReadBucketPipeSyncStatusCoroutine(&sc, sync_pairs[num], sync_status, nullptr);
+  return new RGWReadBucketPipeSyncStatusCoroutine(&sc, sync_pairs[num], sync_status, nullptr, full_status.incremental_gen);
```

Comment:
i don't think anything is initializing this `full_status`. the only other place i see it used is for `RGWRemoteBucketManager::init_sync_status_cr()` which writes a new one with `InitBucketFullSyncStatusCR`

i think `RGWBucketPipeSyncStatusManager::read_sync_status()` needs to read this `full_status` before trying to read any of the shards

Reply:
is this part of our main code path? is this fix needed?

Reply:
its called by `radosgw-admin bucket sync markers`, which is part of the workflow for manual bilog trimming

if bilogs aren't being trimmed automatically (or fast enough), users may have to trim them manually with `radosgw-admin bilog trim` - and doing that safely (i.e. not trimming entries that haven't yet been processed by sync) requires the markers from the peer zone

speaking of, we should probably document this stuff, especially now that we've made it more complicated